'My location' - Selectable/Updating marker representing users current location

DiscoTim opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
This PR adds a selectable and updating location marker representing the users current location.

## Description
<!--- Describe your changes in detail -->
The "My Location" marker is a selectable icon from the same list that the search marker uses, including "Hidden". It can be toggled to follow your location in realtime, or it can be dragged to a desired position. It is separate from and has no effect on the scan marker or scanning. It is completely client side and utilizes the existing locationMarker.

This PR should also fix the bug with not being able to reliably retrieve directions. Directions are based on the 'My Location' marker (visible or not) and are queried with definitive start and end locations.

Fixes bug for search follows location url path for sites running in subdirectories.
Fixes bug for searchMarker will be null on /stats and other pages.

Additionally, some of the wording within the location settings was changed for consistency/clarity.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Currently there is no way to track/update your location within the map, and the scan marker is confusing in that many people assume it is related to your location. With spawn point scanning and larger areas being covered this provides a way to determine/set your location in relation to pokemon..etc..

<!--- If it fixes an open issue, please link to the issue here. -->
https://github.com/PokemonGoMap/PokemonGo-Map/issues/992

https://github.com/PokemonGoMap/PokemonGo-Map/pull/1033

https://github.com/PokemonGoMap/PokemonGo-Map/pull/619

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
 This has been tested on Windows and Android, Firefox and Chrome with both geolocation enabled and disable. (Chrome requires SSL for geolocation outside of localhost)
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):
![nav1](https://cloud.githubusercontent.com/assets/20764373/18224637/c9a727c2-71aa-11e6-9f61-75ec6048ad6a.jpg)
![location2](https://cloud.githubusercontent.com/assets/20764373/18161845/b6708676-7001-11e6-9c27-6db03b3449dd.jpg)
![location3](https://cloud.githubusercontent.com/assets/20764373/18164783/b3ad867e-700f-11e6-9bd8-befed9f8c747.jpg)


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Shibby7986 commented:
cool, I'll try this one out.
I might even have a play with adding sharing the location with other users at some point

Cubkyle commented:
This works perfectly with your #933. If you could add some sort of persistence to it so that I don't have to set the location every time I reopen the page or refresh, it would be perfect.

Shibby7986 commented:
Working fine for me.

DiscoTim commented:
@Cubkyle just to clarify, are you not using geolocation, "Follow my location" option? you are dragging the marker to set it?

I've updated it so the location is persistent.

Cubkyle commented:
@DiscoTim Ah, that's correct. I should have been clearer.

DiscoTim commented:
@Cubkyle okay great, no problem. It should be doing exactly what you want now. thanks for pointing that out.

Cubkyle commented:
Persistence works, and in my testing I found it was per-user, as expected. Thanks!

nunopitbull commented:
I have a question:
I'm running the -nsc and -fl because people were changing their location on map and making my bots get banned. Does this still works with those flags?

DiscoTim commented:
@nunopitbull I've updated with a new commit, and yes it will work with those command switches, and should with any for that matter. 

I think I've fixed the long standing bug with directions as well. I didn't realize that was an existing bug, thanks for pointing that out.

Romfrosk commented:
i also wonder how i can reactivate that people can have their map follow their location without doing searches.

i'm running -SS and don't want them to try something funny :p 

will test this PR and see how it goes.

DiscoTim commented:
@Romfrosk2  this or will do that with the  -nsc and -fl  switches, and they would turn on 'follow my location' , its completely client side, it doesn't effect the search. Each client will have their own location marker. Please let us know how the testing goes.

Shibby7986 commented:
sweet, :) I will turn back on the -nsc :P
my web server instance was not doing any searches anyway but still the icon and options are not needed. for me so its cleaner now

jaquer commented on static/js/map.js:
```diff
@@ -2230,18 +2284,29 @@ $(function () {
+          var baseURL = location.protocol + '//' + location.hostname + (location.port ? ':' + location.port : '')
+          // the search function makes any small movements cause a loop. Need to increase resolution
+          if (getPointDistance(searchMarker.getPosition(), center) > 40) {
+            $.post(baseURL + '/next_loc?lat=' + lat + '&lon=' + lng).done(function () {
```
This is broken for installations running in a subdir. See #619

DiscoTim replied:
@jaquer really surprised this hasn't been fixed yet. it should be fixed now. caught another bug here where the searchMarker will be null on stats and other pages, but this function will still be running.

justsomedudeonthenet commented:
This is an excellent feature I never even realized I wanted!

Tested and is working for me.
